### ISSUE TITLE
warn on identical arguments of assert_eq!()

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3574]
````rust
fn main() {
    let thing_a = 3;
    let thing_b = 4;
    assert_eq!(thing_a, thing_a); // lint here, this should probably be
    // assert_eq!(thing_a, thing_b)
    println!("{} {}", thing_a, thing_b);
}
````

**Comment by Arkweid** [comment]
Is there a good place among the available lint files or need a new one?

**Comment by oli-obk** [comment]
We do have the `eq_op` lint for `x == x`. Maybe we could just not silence it inside the assert family of macros?

**Comment by HarrisonMc555** [comment]
Has anyone picked this up? I could take this one.

**Comment by flip1995** [comment]
You can pick this up!

**Comment by Lythenas** [comment]
@HarrisonMc555 are you still working on this? I already did a similar thing with the `unit_cmp` lint so I could help out.

**Comment by HarrisonMc555** [comment]
@Lythenas I recently started a new job, so I've been pretty busy. If you want to take a stab at it, go for it!

**Comment by Lythenas** [comment]
@HarrisonMc555 It's probably gonna take me a while to get around to this. If you find time feel free to take this otherwise I will eventually do it.
